packages/ra-ui-materialui/src/field/ReferenceArrayField.spec.tsx

```diff
@@ -70,7 +70,7 @@ describe('<ReferenceArrayField />', () => {
                             record={{ id: 123, barIds: [1, 2] }}
                             reference="bar"
                         >
-                            <SingleFieldList>
+                            <SingleFieldList linkType={false}>
```

Comment:
Why is this necessary? `linkType` is optional in `SingleFieldList` props

Reply:
Because the test doesn't define a `ResourceContext`, and `<SingleFieldList>` defaults to creating a link. It's not possible to create a link without a Resource. Before this PR, this caused an empty/invalid link. Now, it throws an error. To avoid this error, we must disable the ink or define a `ResourceContext`.